src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -667,6 +667,23 @@ public static IOperation WalkDownConversion(this IOperation operation)
             return operation;
         }
 
+        /// <summary>
+        /// Walks down consecutive conversion operations that satisfy <paramref name="predicate"/> until an operand is reached that
+        /// either isn't a conversion or doesn't satisfy <paramref name="predicate"/>.
+        /// </summary>
+        /// <param name="operation">The starting operation.</param>
+        /// <param name="predicate">A predicate to filter conversion operations.</param>
+        /// <returns>The first operation that either isn't a conversion or doesn't satisfy <paramref name="predicate"/>.</returns>
+        public static IOperation WalkDownConversion(this IOperation operation, Func<IConversionOperation, bool> predicate)
```

Comment:
Maybe this could be merged with the previous and allow nullable `Func`. Note that I am not convinced whether it brings much.

Reply:
I did end up needing it on [one of my other PRs](https://github.com/dotnet/roslyn-analyzers/pull/4764/files). I can see it being useful in the future, especially for dealing with arguments where you don't want to skip through explicit casts.